### ISSUE TITLE
Access token request is missing client id and secret

### ISSUE THREAD

**Issue by shairyar** [digitalocean/omniauth-digitalocean#25]
I noticed when a request is made to [fetch the access tokens](https://docs.digitalocean.com/reference/api/oauth-api/#2-request-access-token), it is missing `client_id` and `client_secret`. This leads to the following error

```
[2022-10-06T22:33:06.773617 #71279] ERROR -- omniauth: (digitalocean) Authentication failure! invalid_credentials: OAuth2::Error, invalid_request: The request is missing a required parameter, includes an unsupported parameter value, or is otherwise malformed.
{"error":"invalid_request","error_description":"The request is missing a required parameter, includes an unsupported parameter value, or is otherwise malformed."}
```

While debugging, I printed the header payload, and I do not see the `client_id` and `client_secret`

```
{:raise_errors=>true,
 :body=>
  {"grant_type"=>"authorization_code",
   "code"=>"7b52ec4",
   "redirect_uri"=>"https://example.com/users/auth/digitalocean/callback"},
 :headers=>
  {"Content-Type"=>"application/x-www-form-urlencoded",
   "Authorization"=> "Basic kMWMzNWYxNjc1OTUxMzE4"}}
```

Maybe I am overlooking something obvious, but is there any configuration where I need to mention when a request to fetch the access token is sent, it needs to send over those two params as they are required?

**Comment by shairyar** [comment]
Inside the `OmniAuth::Strategies::Digitalocean`, if I add a `token_params` method, that solves the problem. I can then login and sign up using Digital Ocean.

```
def token_params
    options.token_params.merge(options_for("token"))
      .merge(pkce_token_params).merge(client_id: options.client_id)
      .merge(client_secret: options.client_secret)
end
```

I do not know much about OAuth :) So I do not know if there already exists a way to pass that information.

**Comment by excid3** [comment]
@shairyar This is fixed in #26. 👍 

It took me a while to track down what was happening since this goes through several gems.

**Comment by shairyar** [comment]
Nice 👍 thanks for letting me know :)
